reports/2023-project-director-elections-retrospective.md

```diff
@@ -0,0 +1,112 @@
+
+One measure is the number of calendar days that it took to do the process.
+If we go by the blog posts that announced the beginning of the process and the end of the process, this took from August 30 until October 19, or 50 days.
+We needed to allow at least two weeks (14 days) for the nomination phase, and 10 days for the feedback phase, so the absolute minimum possible time would be 24 days.
+In practice, it's helpful to have some slack time between these phases, and we also needed time to coordinate the election meeting itself.
+Note that these 50 days do not include the time to design and plan for the process ahead of time.
+In the future, there should be less preplanning needed because we will not be designing the process from scratch.
+
+One reasons slack time in the schedule was helpful was that we neglected to plan time for nominees to consider whether to accept their nomination, and also time to confirm with the candidates who were selected.
+Going forward we should allow for this, because being willing to serve as a Project Director is a consequential decision that deserves adequate time to consider.
+
+Another measure of time efficiency is the time it took the people involved in the process to coordinate and do each of the steps.
+The role of Project Director Facilitator took a significant amount of time.
+It would be nice to reduce the time requirement, perhaps by subdividing the role.
+
+Relating to scheduling, we also need to consider scheduling constraints outside the election process itself.
+For example, candidate nominations closed on the last day of RustConf, and we were working on the blog post announcing the results of the election during the week of EuroRust.
+Going forward, we should consider major conferences of broad interest to the community and avoid scheduling election milestones during those times.
+
+It's not clear how best to think about time efficiency.
+In some ways, we spent a reasonable amount of time.
+Still, the process seemed more time-consuming than expected.
+It's likely that this is primarily the result of under-estimating the time it takes to do Project Director Elections well.
+
+### Incorporate Input from a Wide Range of Stakeholders Across the Entire Rust Project
+
+We feel overall we did well on this goal.
+The steps in the process were widely communicated.
+The nomination process being done per-team helped to ensure we had good coverage across the Rust project.
+During the feedback phase, we felt like we heard from a variety of people in the project.
+
+### Ensure a Diverse Pool of Qualified Candidates for Selection
+
+We had nine qualified candidates nominated.
+Along several dimensions, diversity was rather low.
+It's hard to quantify this exactly because we did not collect demographic information about the nominees.
+
+While not usually considered diversity, given the impact of employment limits, we should perhaps consider having nominees from a diverse set of employers as part of this goal.

```

Comment:
This will be hard to change since it's in the bylaws, but as I watched the process unfold I agree it felt artificially limiting at times, particularly for big corps where two candidates may be in completely different parts of the company. At the same time, I don't think it would be acceptable to have more than three of five project directors from the same company.

I agree the limit speaks to diversity somewhat, but I think it exists mostly to limit the *influence* of particular actors. If the hard limit gets raised I think we should be more careful to consider this during selection. Even under the current limit, it would be weird to select two candidates who are on the same team and work closely together.

This suggests another framing to me: While under the current rules we tend to think of corporations as monoliths, in reality they are made up of people with diverging interests. In considering candidates from the same company, it would be helpful to consider how much of the reporting chain they share in common, and how much influence someone in the overlap might want to wield over Rust. Ideally both formal and informal "reporting chains" would be considered. Of course this can be hard to do effectively, which is why a hard rule is easiest to enforce.